data-processing/common/parse_equation.py

```diff
@@ -64,23 +66,54 @@ def get_symbols(mathml: str) -> List[Symbol]:
             if index is not None:
                 characters.append(index)
 
-        # Clean the MathML of our annotations. Do this cleaning on a clone of the node, so that the
-        # annotations are available for processing later nodes.
-        node_clone = copy.copy(symbol_node)
-        _remove_s2_annotations(node_clone)
-        for descendant in node_clone.descendants:
-            _remove_s2_annotations(descendant)
+        # Group consecutive character symbols only if they are a child of the mrow tag.
+        # The .has_attr('s2:end') is necessary as not every symbol is annotated by katex
+        if symbol_node.has_attr('s2:end') and symbol_node.parent.name in VALID_PARENT_TAG and symbol_node.name in CHARACTER_TAGS:
+            if len(consecutive_chars) > 0 and not consecutive_chars[-1]['s2:end'] == symbol_node['s2:start']:
```

Comment:
Do we need this constraint that checks that `s2:end` doesn't align with the previous character's `s2:start`? Can we just look for `mi` nodes that follow one after the other?

As an example, I wonder if this condition would prevent characters from getting grouped for the following LaTeX:

```latex
$R e L U$        % spaces in between characters
${R}{e}{L}{U}$   % would appear almost if not exactly the same as $ReLU$
```

where R, e, L, and U might (though I don't know for sure) have non-overlapping start and end indices. I don't know enough about how KaTeX parses these to know whether the MathML would have `start` and `end` numbers compatible with your algorithm. It's worth a test!

Reply:
Gotcha, let me test it out. I originally ran into a lot of trouble trying to use .sibling_next and what not but maybe I can take another crack at it.

Reply:
Looks like KaTeX spits out the same thing for all three so we should be good.